Application/src/view/visuals/component/ClassRelationship.java

```diff
@@ -0,0 +1,257 @@
+        int arrowSize = 15; // Initial arrow size is 15 but doubled later for diagonal lines
+        int arrowAngle = 0; // used to rotate the arrow to 8 different directions
+
+        // the following 8 if statements are used to rotate the arrow according to vertical and horizontal distance,
+        // note: looking for a better solution with less complexity
+
+        // point east
+        if (XDistance > 0 && YDistance == 0) {
+            arrowAngle += 90;
+        }
+
+        // point west
+        else if (XDistance < 0 && YDistance == 0) {
+            arrowAngle += 270;
+        }
+
+        // point north
+        else if (XDistance == 0 && YDistance > 0) {
+            arrowAngle += 180;
+        }
+
+        // point south
+        else if (XDistance == 0 && YDistance < 0) {
+            arrowAngle += 0;
+        }
+
+        // point south east
+        else if (XDistance > 0 && YDistance > 0) {
+            arrowAngle += 135;
+        }
+
+        // point north east
+        else if (XDistance > 0 && YDistance < 0) {
+            arrowAngle += 45;
+        }
+
+        // point north west
+        else if (XDistance < 0 && YDistance < 0) {
+            arrowAngle += 315;
+        }
+
+        // point south west
+        else if (XDistance < 0 && YDistance > 0) {
+            arrowAngle += 225;
+        }
+
+        // negates the operand
+        if (XDistance < 0) {
+            offsetIncrementX *= (-1);
+        }
+        // negates the operand
+        if (YDistance < 0) {
+            offsetIncrementY *= (-1);
+        }
+
+        // To solve the issue with all the different cases of linking the classes, we used Bresenham’s Line Algorithm
+        // http://csunplugged.org/wp-content/uploads/2014/12/Lines.pdf
+        int A, B, P;
+
+        // This is used to find out whether to use the Y axis or the X axis as a parameter for the for loop
+        // this avoids having 2 loops
+        int stepsParameter;
+
+        /// Using the Bresenham’s Line Algorithm on X axis as reference
+        if (Math.abs(XDistance) > Math.abs(YDistance)) {
+            stepsParameter = Math.abs(XDistance);
+
+            A = 2 * Math.abs(YDistance);
+            B = A - 2 * Math.abs((XDistance));
+            P = A - Math.abs((XDistance));
+            // decreasing the gap to show arrow correctly when it is a horizontal line
+            if (YDistance == 0) {
+                gap += -2;
+            }
+            // decreasing the gap to show arrow correctly when it is a horizontal line pointing right
+            if (XDistance > 0) {
+                gap += -1;
+            }
+            // increasing the gap to show arrow correctly when it is a horizontal line pointing left
+            if (XDistance < 0) {
+                gap += 1;
+            }
+        }
+
+        // Using the Bresenham’s Line Algorithm on the Y axis as reference
+        else {
+            stepsParameter = Math.abs(YDistance);
```

Comment:
Make it into a method?

Reply:
i am also using the if else statement as a reference to where to change the gap variable.

Reply:
Well you should be able to put that into the method

Reply:
+1